static/app/stores/Commiters/CommittersContext.tsx

```diff
@@ -0,0 +1,279 @@
+    committers: Committer[];
+    eventId: string;
+    organizationSlug: string;
+    projectSlug: string;
+  };
+  type: 'add committers';
+};
+
+export type CommittersState = Record<
+  string,
+  {
+    committers: Committer[];
+    committersError: boolean;
+    committersLoading: boolean;
+  }
+>;
+
+export type CommittersAction =
+  | ResetAction
+  | AddCommitters
+  | StartLoadingCommitters
+  | SetCommittersError;
+
+export function committersReducer(state, action: CommittersAction): CommittersState {
+  switch (action.type) {
+    case 'add committers': {
+      const key = makeCommitterStoreKey({
+        organizationSlug: action.payload.organizationSlug,
+        projectSlug: action.payload.projectSlug,
+        eventId: action.payload.eventId,
+      });
+
+      return {
+        ...state,
+        [key]: {
+          committers: action.payload.committers,
+          committersLoading: false,
+          committersError: undefined,
+        },
+      };
+    }
+
+    case 'start loading': {
+      const key = makeCommitterStoreKey({
+        organizationSlug: action.payload.organizationSlug,
+        projectSlug: action.payload.projectSlug,
+        eventId: action.payload.eventId,
+      });
+      return {
+        ...state,
+        [key]: {
+          committers: [],
+          committersLoading: true,
+          committersError: undefined,
+        },
+      };
+    }
+    case 'set error': {
+      const key = makeCommitterStoreKey({
+        organizationSlug: action.payload.organizationSlug,
+        projectSlug: action.payload.projectSlug,
+        eventId: action.payload.eventId,
+      });
+
+      return {
+        ...state,
+        [key]: {
+          committers: [],
+          committersLoading: false,
+          committersError: true,
+        },
+      };
+    }
+    case 'reset': {
+      return {};
+    }
+    default: {
+      exhaustive(action);
+      return state;
+    }
+  }
+}
+
+export const CommittersContext = createContext<
+  [CommittersState, React.Dispatch<CommittersAction>] | null
+>(null);
+
+interface CommittersContextProviderProps {
+  children: React.ReactNode;
+  initialState?: CommittersState;
+}
+
+export function CommittersProvider(props: CommittersContextProviderProps) {
+  const contextValue = useReducer(committersReducer, props.initialState ?? {});
+
+  return (
+    <CommittersContext.Provider value={contextValue}>
+      {props.children}
+    </CommittersContext.Provider>
+  );
+}
+
+export function useCommitters(): [CommittersState, React.Dispatch<CommittersAction>] {
+  const context = useContext(CommittersContext);
+
+  if (!context) {
+    throw new Error('useCommitters called outside of CommittersContext.Provider');
+  }
+
+  return context;
+}
+
+interface RequiredWithCommittersProps {
+  event: Event;
+  organization: Organization;
+  project: Project | AvatarProject;
+  group?: Group;
+}
+
+export interface WithCommittersProps {
+  committers: Committer[] | undefined;
```

Comment:
```suggestion
  committers?: Committer[];
```

Reply:
I think I prefer the type above. We want to be sure that our hoc always passes committers prop altho the value can indeed be undefined. You can see the difference of the two here https://codesandbox.io/s/kind-silence-vi1jvy?file=/src/App.tsx:0-278